test/Sia.Gateway.Tests/Filters/EventFiltersTests.cs

```diff
@@ -20,7 +20,7 @@ public void Filter_WhenPassedEmptyQueryable_ReturnsEmptyQueryable()
             var testInput = new List<Event>().AsQueryable();
 
 
-            var result = serviceUnderTest.Filter(testInput);
+            var result = testInput.WithFilter(serviceUnderTest);
```

Comment:
Each test that's invoking `WithFilter` is effectively testing behavior for both `FilterExtensions.WithFilter` and the `EventFilters` object being passed to it. 

Testing the behavior of multiple objects together like this turns this unit test into something more like an integration test (definitions of integration test vary). The net effect is that the testing code becomes more complicated and makes the project hard to maintain and extend: 
* If you change the interface for `WithFilter` you'll break tests that are really only supposed to be validating that the `EventFilters` object works properly. 
* If the two modules have multiple return paths that need to be tested then the test matrix can get really complicated (not a problem in the case of `WithFilter`, but if `WithFilter` had conditionals or a try/catch then this situation would probably apply)

The solution here is to simplify these tests to match the new tests you wrote. In this case, something like:
```
        [TestMethod]
        public void DoesNotMatch_OnEmptyEventFilters()
        {
            var serviceUnderTest = new EventFilters();
            var testInput = new Event()
            {
                EventTypeId = 2,
            };

            var result = serviceUnderTest.IsMatchFor(testInput);

            Assert.IsFalse(result);
        }
```

Reply:
Added tests for individual filtering cases that test against IsMatchFor directly. Moved 'integration' tests to a separate filterExtensionsTests file; we can get rid of that separately if we find it's not valuable.